examples/image-classifier-finetuning/train.py

```diff
@@ -0,0 +1,181 @@
+    'net__batch_size': 16,
+    'net__optimizer': torch.optim.AdamW,
+    'net__lr': 2e-4,
+    'net__optimizer__weight_decay': 0.0,
+    'net__iterator_train__shuffle': True,
+    'net__train_split': False,
+}
+
+
+def get_data():
+    ds = load_dataset('beans')
+
+    X_train = ds['train']['image']
+    y_train = np.array(ds['train']['labels'])
+
+    X_valid = ds['validation']['image']
+    y_valid = np.array(ds['validation']['labels'])
+
+    return X_train, X_valid, y_train, y_valid
+
+
+class FeatureExtractor(BaseEstimator, TransformerMixin):
+    """Image feature extractor
+
+    Parameters
+    ----------
+    model_name : str (default='google/vit-base-patch32-224-in21k')
+      Name of the feature extractor on Hugging Face Hub.
+
+    device : str (default='cuda')
+      Computation device, typically 'cuda' or 'cpu'.
+
+    """
+    def __init__(
+            self,
+            model_name='google/vit-base-patch32-224-in21k',
+            device='cuda',
+    ):
+        self.model_name = model_name
+        self.device = device
+
+    def fit(self, X, y=None, **fit_params):
+        self.extractor_ = ViTFeatureExtractor.from_pretrained(
+            self.model_name, device=self.device,
+        )
+        return self
+
+    def transform(self, X):
+        return self.extractor_(X, return_tensors='pt')['pixel_values']
+
+
+class VitModule(nn.Module):
+    """Vision transformer module
+
+    Parameters
+    ----------
+    model_name : str (default='google/vit-base-patch32-224-in21k')
+      Name of the feature extractor on Hugging Face Hub.
+
+    num_classes : int (default=3)
+      Number of target classes to classify.
+
+    """
+    def __init__(
+            self,
+            model_name='google/vit-base-patch32-224-in21k',
+            num_classes=3,
+    ):
+        super().__init__()
+        self.model = ViTForImageClassification.from_pretrained(
+            model_name, num_labels=num_classes
+        )
+
+    def forward(self, X):
+        X = self.model(X)
+        return X.logits
+
+
+def make_lr_lambda(num_warmup_steps, num_training_steps, last_epoch=-1):
+    # Since this returns a local function, the resulting net cannot be pickled,
+    # unless we call net.trim_for_prediction(). If you want to be able to pickle
+    # this, rewrite this function as a callable class.
+    def lr_lambda(current_step: int):
+        if current_step < num_warmup_steps:
+            return float(current_step) / float(max(1, num_warmup_steps))
+        return max(
+            0.0, float(num_training_steps - current_step) / float(
+                max(1, num_training_steps - num_warmup_steps))
+        )
+    return lr_lambda
+
+
+def get_model(num_classes, lr_lambda):
+    pipe = Pipeline([
+        ('feature_extractor', FeatureExtractor()),
+        ('net', NeuralNetClassifier(
+            VitModule,
+            callbacks=[
+                LRScheduler(LambdaLR, lr_lambda=lr_lambda),
+                ProgressBar(),
+            ],
```

Comment:
Should `num_classes` be passed to the module?

```suggestion
            ],
            module__num_classes=num_classes
```